v2/api/network/customizations/private_endpoints_extensions.go

```diff
@@ -60,3 +67,109 @@ func (extension *PrivateEndpointExtension) PostReconcileCheck(
 
 	return extensions.PostReconcileCheckResultSuccess(), nil
 }
+
+func (extension *PrivateEndpointExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	// This has to be the current hub storage version. It will need to be updated
+	// if the hub storage version changes.
+	endpoint, ok := obj.(*network.PrivateEndpoint)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T, expected *network.PrivateEndpoint", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = endpoint
+
+	hasIpConfiguration := hasConfigMaps(endpoint)
+	if !hasIpConfiguration {
+		log.V(Debug).Info("no configmap retrieval to perform as operatorSpec is empty")
+		return nil, nil
+	}
+
+	if endpoint.Status.NetworkInterfaces == nil || len(endpoint.Status.NetworkInterfaces) == 0 {
+		log.V(Debug).Info("no configmap retrieval to perform as there are no NetworkInterfaces attached")
+		return nil, nil
+	}
+
+	if endpoint.Status.NetworkInterfaces[0].Id == nil {
```

Comment:
Is the default network interface always `[0]`? Or do you need to scan them for an `.IsDefault` property or something?

Reply:
I added a scan for this at the IPConfiguration level but left the assumption that there's only 1 NIC because as far as I can tell there's no dual-NIC'ed PEs.